tests/rptest/tests/cluster_recovery_test.py

```diff
@@ -286,6 +286,91 @@ def test_admin_recovery(self):
         self.redpanda._admin.initialize_cluster_recovery()
         wait_until(self._cluster_recovery_complete, timeout_sec=60, backoff_sec=1)
 
+    @cluster(num_nodes=1)
+    def test_admin_recovery_uuid_override(self):
+        rpk = RpkTool(self.redpanda)
+
+        self.redpanda._admin.patch_cluster_config(
+            {
+                "cloud_storage_cluster_name": "test-my-cluster",
+            }
+        )
+        self._wait_for_metadata_upload()
+        initial_cluster_uuid = self.redpanda._admin.get_cluster_uuid()
+        self.logger.debug(f"Initial cluster uuid: {initial_cluster_uuid}")
+
+        self.logger.info(
+            "Recreating the cluster with the same name but do not attempt recovery"
+        )
+        self._stop_and_recreate_cluster(
+            {
+                "cloud_storage_cluster_name": "test-my-cluster",
+            }
+        )
+        self._wait_for_metadata_upload()
+
+        self.logger.info("Verifying that no recovery happened")
+        topics_on_cluster = set(rpk.list_topics())
+        assert len(topics_on_cluster) == 0, (
+            f"Expected no topics to be restored but have: {topics_on_cluster}"
+        )
+
+        self.logger.info("Recreate the cluster and attempt recovery from the first one")
+        self._stop_and_recreate_cluster(
+            {
+                # Do not set the name yet, we want to validate that recovery fails without it first.
+                "cloud_storage_cluster_name": None,
+            }
+        )
+        self.logger.info(
+            "Verifying that recovery without name fails if uuid override is provided, name is not set but bucket contains multiple clusters"
+        )
+        with pytest.raises(requests.exceptions.HTTPError) as excinfo:
+            self.redpanda._admin.initialize_cluster_recovery(
+                cluster_uuid_override=initial_cluster_uuid
+            )
+        assert excinfo.value.response.status_code == 400, (
+            f"Status: {excinfo.value.response.status_code}"
+        )
+        assert (
+            excinfo.value.response.json()["message"]
+            == "Cluster is misconfigured for recovery. Check logs for details."
+        ), excinfo.value.response.json()["message"]
+
+        self.logger.info("Set a different cluster name and attempt to recovery")
+        self.redpanda._admin.patch_cluster_config(
+            {
+                "cloud_storage_cluster_name": "some-random-name",
+            }
+        )
+        with pytest.raises(requests.exceptions.HTTPError) as excinfo:
+            self.redpanda._admin.initialize_cluster_recovery(
+                cluster_uuid_override=initial_cluster_uuid
+            )
+        assert excinfo.value.response.status_code == 400, (
+            f"Status: {excinfo.value.response.status_code}"
+        )
+        assert (
+            excinfo.value.response.json()["message"]
+            == "Cluster is misconfigured for recovery. Check logs for details."
```

Comment:
Not suggesting an immediate change, but this makes me wonder if we should also be propagating the error (or at least some more specific details) to the admin endpoint response.

Reply:
Don't disagree but tried to stick to existing conventions for error code propagation. Don't want to spend time here and now for error propagation. Made sure there are comprehensive logs about recovery process though. Net change is positive imho.